tree/treeplayer/inc/ROOT/TDFActionHelpers.hxx

```diff
@@ -427,16 +429,19 @@ class SnapshotHelper {
 
 public:
    SnapshotHelper(std::string_view filename, std::string_view dirname, std::string_view treename,
-                  const ColumnNames_t &bnames)
-      : fOutputFile(TFile::Open(std::string(filename).c_str(), "RECREATE")), fBranchNames(bnames)
+                  const ColumnNames_t &bnames, SnapshotOptions const &options)
+      : fOutputFile(TFile::Open(std::string(filename).c_str(), options.mode.c_str(), /*ftitle=*/"", options.compress)), fBranchNames(bnames)
    {
       if (!dirname.empty()) {
          std::string dirnameStr(dirname);
          fOutputFile->mkdir(dirnameStr.c_str());
          fOutputFile->cd(dirnameStr.c_str());
       }
       std::string treenameStr(treename);
-      fOutputTree.reset(new TTree(treenameStr.c_str(), treenameStr.c_str(), /*split=*/99, /*dir=*/fOutputFile.get()));
+      fOutputTree.reset(new TTree(treenameStr.c_str(), treenameStr.c_str(), options.splitlevel, /*dir=*/fOutputFile.get()));
+
+      if (options.autosave)
+        fOutputTree->SetAutoSave(options.autosave);
```

Comment:
`autosave`/`autoflush` support negative values -- why the `if`?

Reply:
Because the default in `SnapshotOptions` is 0. Therefore, if the user did not specify it, we don't set it in the TTree and automatically get whatever the default is there. Also, setting auto save to 0 probably has bad consequences :-)

Reply:
but what if the user specifies a negative value?